index.js

```diff
@@ -169,7 +169,11 @@ module.exports = {
    */
   _getIncludesForInRepoAddonDirectories() {
     return this._findInRepoAddons().reduce((acc, addon) => {
-      let addonDir = path.join(this.project.root, 'lib', addon.name);
+      let addonDir =
+        typeof addon.root === 'string'
+          ? addon.root
+          : path.join(this.project.root, 'lib', addon.name);
```

Comment:
I agree, this seems super odd / not good. Assuming that an addon is in `lib/some-name` is definitely broken.